python/benchmark/gen_data_distributed.py

```diff
@@ -479,13 +479,14 @@ def make_regression_udf(iter: Iterable[pd.DataFrame]) -> Iterable[pd.DataFrame]:
                     logging.warning("cupy import failed; falling back to numpy.")
 
             partition_index = pyspark.TaskContext().partitionId()
+            my_seed = partition_seeds[partition_index % len(partition_seeds)]
```

Comment:
Good finding on this, but is there a way to guarantee unique seeds across the partitions?   With this, we could have two partitions_indexes that differ by a multiple of partition_seeds and then they would have the same seed.

Reply:
Maybe we can set task id as the random seed. However, the generated dataset is non-deterministic because the set of task ids changes in different runs. 

Will think about if there is a way to guarantee both unique seeds and deterministic dataset.

Reply:
Will you be revising this PR?

Reply:
Revised to "partition_seed = global_random_state + partition id".